server/src/main/java/org/elasticsearch/snapshots/SnapshotShardsService.java

```diff
@@ -452,14 +424,13 @@ private void syncShardStatsOnNewMaster(ClusterChangedEvent event) {
                                 // but we think the shard is done - we need to make new master know that the shard is done
                                 logger.debug("[{}] new master thinks the shard [{}] is not completed but the shard is done locally, " +
                                              "updating status on the master", snapshot.snapshot(), shardId);
-                                notifySuccessfulSnapshotShard(snapshot.snapshot(), shardId, localNodeId, masterNode);
+                                notifySuccessfulSnapshotShard(snapshot.snapshot(), shardId, masterNode);
```

Comment:
All this passing down of `masterNode` is different from the `7.x/master` version since we need the Bwc path in the status update sending below.